pysat/tests/test_instruments.py

```diff
@@ -146,97 +149,66 @@ def test_inst_max_latitude(self, inst_dict):
-
-        with warnings.catch_warnings(record=True) as war:
-            try:
-                itc.initialize_test_inst_and_date({})
-            except KeyError:
-                # empty dict produces KeyError
-                pass
-
-        self.warn_msgs = ["`initialize_test_inst_and_date` has been moved to"]
-        self.warn_msgs = np.array(self.warn_msgs)
-
-        # Ensure the minimum number of warnings were raised
-        assert len(war) >= len(self.warn_msgs)
-
-        # Test the warning messages, ensuring each attribute is present
-        testing.eval_warnings(war, self.warn_msgs)
-        return
-
-    def test_old_pytest_mark_presence(self):
-        """Test that pytest mark is backwards compatible."""
-
-        n_args = len(InstLibTests.test_load.pytestmark)
-        mark_names = [InstLibTests.test_load.pytestmark[j].name
-                      for j in range(0, n_args)]
-
-        assert "download" in mark_names
-
-    @pytest.mark.parametrize("inst_module", ['pysat_testing2d',
-                                             'pysat_testing_xarray',
-                                             'pysat_testing2d_xarray'])
-    def test_deprecated_instruments(self, inst_module):
-        """Check that instantiating old instruments raises a DeprecationWarning.
+    @pytest.mark.second
+    @pytest.mark.parametrize("clean_level", ['clean', 'dusty', 'dirty'])
+    @pytest.mark.parametrize("change", [True, False])
+    @pytest.mark.parametrize('warn_type', ['logger', 'warning', 'error',
+                                           'mult'])
+    @pytest.mark.parametrize("inst_dict", instruments['download'])
+    def test_clean_with_warnings(self, clean_level, change, warn_type,
+                                 inst_dict, caplog):
+        """Run `test_clean_warn` with different warning behaviours.
 
         Parameters
         ----------
-        inst_module : str
-            name of deprecated module.
+        clean_level : str
+            Cleanliness level for loaded instrument data; must run the clean
+            routine (not include 'none').
+        change : bool
+            Specify whether or not clean level should change.
+        warn_type : str
+            Desired type of warning or error to be raised.
+        inst_dict : dict
+            One of the dictionaries returned from
+            `InstLibTests.initialize_test_package` with instruments to test.
 
         """
+        # Set the default values
+        warn_level = {'logger': 'WARN', 'warning': UserWarning,
+                      'error': ValueError}
+        warn_msg = 'Default warning message'
+        if change:
+            final_level = 'none' if clean_level == 'clean' else 'clean'
+        else:
+            final_level = clean_level
+
+        # Construct the expected warnings
+        if warn_type == 'mult':
+            # Note that we cannot test errors along with other warnings
```

Comment:
See #1178 for combo of error and warning test. Could be noted as an issue.

Reply:
Could you write this up as an issue?

Reply:
Issue #1184 created